frontend/ui/uimanager.lua

```diff
@@ -714,7 +707,9 @@ function UIManager:handleInput()
 
     -- delegate input_event to handler
     if input_event then
-        self:_resetAutoSuspendTimer()
+        if input_event.handler ~= "onInputError" then
+            self:broadcastEvent(Event:new("InputEvent"))
```

Comment:
broadcasting every input event too all widgets seems too expensive. another approach to do is is to implement some kind of a hook system where the plugin can register a callback. then have the uimanager check the list of hooks and call them here. basically something similar to the menu entry registration code path.